SDWebImageFLPlugin/Classes/FLAnimatedImageBridge/SDWebImageFLCoder.m

```diff
@@ -0,0 +1,74 @@
+/*
+ * This file is part of the SDWebImage package.
+ * (c) Olivier Poitrey <rs@dailymotion.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+#import "SDWebImageFLCoder.h"
+#import "FLAnimatedImageView+WebCache.h"
+
+@implementation SDWebImageFLCoder
+
++ (SDWebImageFLCoder *)sharedCoder {
+    static SDWebImageFLCoder *coder;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        coder = [[SDWebImageFLCoder alloc] init];
+    });
+    return coder;
+}
+
+- (BOOL)canDecodeFromData:(NSData *)data {
+    return ([NSData sd_imageFormatForImageData:data] == SDImageFormatGIF);
+}
+
+- (UIImage *)decodedImageWithData:(NSData *)data options:(SDImageCoderOptions *)options {
+    SDWebImageContext *context = options[SDImageCoderWebImageContext];
+    NSString *operationKey = context[SDWebImageContextSetImageOperationKey];
+    
+    // Check if image request come from `FLAnimatedImageView`
+    if ([operationKey isEqualToString:NSStringFromClass(FLAnimatedImageView.class)]) {
+        // Parse args
+        BOOL predrawingEnabled = YES;
+        if (context[SDWebImageContextPredrawingEnabled]) {
+            predrawingEnabled = [context[SDWebImageContextPredrawingEnabled] boolValue];
+        }
+        NSUInteger optimalFrameCacheSize = 0;
+        if (context[SDWebImageContextOptimalFrameCacheSize]) {
+            optimalFrameCacheSize = [context[SDWebImageContextOptimalFrameCacheSize] unsignedIntegerValue];
+        }
+        // Create FLAnimatedImage
+        FLAnimatedImage *animatedImage = [[FLAnimatedImage alloc] initWithAnimatedGIFData:data optimalFrameCacheSize:optimalFrameCacheSize predrawingEnabled:predrawingEnabled];
```

Comment:
The same.

Reply:
We don't support subclass.
```
@interface SubclassOfFLAnimatedImageView : FLAnimatedImageView
@end
```

Reply:
Oops, this line don't needs to consider subclass. Just [this line](https://github.com/SDWebImage/SDWebImageFLPlugin/pull/2/files#r205999019).

Reply:
Added one commit to support subclass.

Anyway, most of users does not use subclass of `FLAnimatedImageView`. However, this is possible and support on it is easy

Reply:
So I'm sensitive to these things. The `class` would always better than direct `String` compare. 😂